### PR TITLE
Add ability to ignore limit switches when climbing

### PR DESCRIPTION
**Pull request by JohnGilb**
Adds an `AscendDangerouslyCommand` that will run the climber even if limit switches are pressed.

### CODE REVIEW

File: Competition2017/src/competition/operator_interface/OperatorCommandMap.java [hunk 1]
```diff
@@ -52,14 +52,15 @@ public void setupMyCommands(
...
+            DescendClimbingCommand descend,
+            AscendSafelyCommand ascend,
+            AscendDangerouslyCommand ascendDangerously,
             RopeAlignerCommand aligner)   
     {
         oi.controller.getXboxButton(XboxButton.X).whileHeld(ascend);
         
         oi.controller.getXboxButton(XboxButton.Start).whileHeld(aligner);
-        
-        oi.operatorPanelButtons.getifAvailable(4).whileHeld(ascend);
```
Comment (WasabiFan) [inline Competition2017/src/competition/operator_interface/OperatorCommandMap.java]: Replacing a panel button with an Xbox button? Are we no longer using the panel button?

File: Competition2017/src/competition/subsystems/climbing/ClimbingSubsystem.java [hunk 2]
```diff
@@ -56,6 +58,14 @@ public void descend() {
     public void setBrake(boolean brakeOn) {
         brakeSolenoid.set(brakeOn);
     }
+    
+    public boolean isLimitSwitchPressed() {
+        return climbingMotor.isForwardLimitSwitchClosed();
+    }
+    
+    public void setEnableSafeties(boolean isEnabled) {
+        climbingMotor.enableLimitSwitches(isEnabled, isEnabled);
```
Comment (WasabiFan) [inline Competition2017/src/competition/subsystems/climbing/ClimbingSubsystem.java]: Earlier, the reverse limit switch is configured to be disabled universally; I think we want to do that here as well.

**Review by WasabiFan — CHANGES_REQUESTED** [review]
We should make sure to test this and confirm that the Talons process the limit switch enable/disable in such a way that they take effect immediately.

File: Competition2017/src/competition/subsystems/climbing/ClimbingSubsystem.java (hunk 2 above)
Comment (JohnGilb) [reply Competition2017/src/competition/subsystems/climbing/ClimbingSubsystem.java]: Yes, we should be consistent.

File: Competition2017/src/competition/operator_interface/OperatorCommandMap.java (hunk 1 above)
Comment (JohnGilb) [reply Competition2017/src/competition/operator_interface/OperatorCommandMap.java]: Ah, didn't mean to also replace the panel. Thanks for noticing.

**Pull request closed without merging**